presto-docs/src/main/sphinx/release/release-0.199.rst

```diff
@@ -0,0 +1,66 @@
+=============
+Release 0.199
+=============
+
+General Changes
+---------------
+
+* Allow users to create views for their own use when they do not have permission
+  to grant others access to the underlying tables or views. To enable this,
+  creation permission is now only checked at query time, not at creation time,
+  and the query time check is skipped if the user is the owner of the view.
+* Add support for spatial left join.
+* Add :func:`hmac_md5`, :func:`hmac_sha1`, :func:`hmac_sha256`, and :func:`hmac_sha512` functions.
+* Add :func:`array_sort` function that takes a lambda as a comparator.
+* Add :func:`line_locate_point` geospatial function.
+* Add support for ``ORDER BY`` clause in aggregations for queries that use grouping sets.
+* Expand grouped execution support to ``GROUP BY`` and ``UNION ALL``, making it possible
+  to execute aggregations with less peak memory usage.
+* Change the signature of ``round(x, d)`` and ``truncate(x, d)`` functions so that
+  ``d`` is of type ``INTEGER``. Previously, ``d`` could be of type ``BIGINT``.
+  This behavior can be reverted with the ``deprecated.legacy-round-n-bigint`` config option
+  or the ``legacy_round_n_bigint`` session property.
+* Add support for installing Presto server RPM to machines with OpenJDK.
+* Optimize the :func:`ST_Intersection` function for rectangles aligned with coordinate axes
+  (e.g., polygons produced by the :func:`ST_Envelope` and :func:`bing_tile_polygon` functions).
+* Finish joins early when possible if one side has no rows. This happens for
+  either side of an inner join, for the left side of a left join, and for the
+  right side of a right join.
+* Improve planning performance when evaluating partition candidates.
```

Comment:
We still need this to be updated.

Reply:
Yes, I already pinged @kokosing about it. We need his input to understand the details.

Reply:
It is about `AddExchanges#visitTableScan`, where we pass `LayoutConstraintEvaluator::isCandidate` in `Constraint` to `ConnectorMetadata::getTableLayouts`. 

I had a case when table had many string partitions which caused `TableScanNode::currentConstaint` to be big. As a result in `AddExchanges` had to evaluate big predicate for every partitions once again in `LayoutConstraintEvaluator::isCandidate`. That predicate was in form `a LIKE 'pattern' AND a IN (CAST('value1' AS STRING), ..., huge list of casts)` (somehow predicate was not simplified). Most of partitions were filtered out on the first conjunct, but second was also evaluated. So the change was about short-circuit such predicate evaluation and finish predicate evaluation on first failed conjunct. 
It helped in my case, but it is not a general fix for above problem. It is rather simple generic optimization here.

Reply:
@kokosing Thanks for the explanation. Can you provide a release note for this that explains it in general terms for the engine? (see other examples of how we do this)

We can add a "For example, this improves XXX for the Hive connector" at the end

Reply:
Maybe `Improve planning performance when querying partitioned tables`?

Reply:
What about `Improve predicate evaluation performance during predicate pushdown in planning.`